udao/optimization/tests/soo/test_mogd.py

```diff
@@ -159,21 +163,32 @@ def test_solve(
         assert optimal_vars == expected_vars
 
     @pytest.mark.parametrize(
-        "variable, expected_obj, expected_variable, batch_size",
+        "strict_rounding, variable, batch_size, expected_obj, expected_vars",
         [
-            (co.FloatVariable(0, 24), 148.17662, {"cores": 16.2}, 1),
-            (co.FloatVariable(0, 24), 148.17662, {"cores": 16.2}, 16),
-            (co.IntegerVariable(0, 24), 150, {"cores": 16}, 1),
-            (co.IntegerVariable(0, 24), 150, {"cores": 16}, 16),
+            (
+                True,
+                co.IntegerVariable(0, 24),
+                1,
+                200,
+                {"cores": 12},
```

Comment:
Do you know why optimization "fails" in this case?

Reply:
The rounding at the end of iteration can cancel out the gradient descent in some unluck cases